CONTRIBUTING.rst

```diff
@@ -735,27 +735,135 @@ extension, follow coding guidelines as established by `Sphinx-Gallery
+module for more details on how to create a suitable dataset loader.
+
+Using the dataset loaders this way will enable any metadata for the new
+dataset to be automatically included in the :ref:`dataset_gallery`.
+
+New datasets should also provide a sample plot of the dataset in the
+``Examples`` section of the docstring of the new function, e.g.:
+
+.. code::
+
+   def download_my_new_mesh(load=True):
+      """Download my new mesh.
+
+      Examples
+      --------
+      >>> from pyvista import examples
+      >>> dataset = examples.download_my_new_mesh()
+      >>> dataset.plot()
+      """
+
+In addition, the output plot image must be manually added to
+``pyvista/doc/source/images/dataset-gallery`` for it to be used as a
+thumbnail in the :ref:`dataset_gallery`. The file name should follow the
+same convention as existing files in that directory.
+
+Extending the Dataset Gallery
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If you have multiple related datasets to contribute, or would like to
+group any existing datasets together which share similar properties,
+the :ref:`dataset_gallery` can easily be extended to feature these
+datasets in a new `card carousel<https://sphinx-design.readthedocs.io/en/latest/cards.html#card-carousels>`_.
+
+For example, to add a new ``Instrument`` dataset category to :ref:`dataset_gallery_category`
+featuring two datasets (1) :func:`pyvista.examples.download_guitar` and (2)
+:func:`pyvista.examples.download_trumpet`, complete the following steps:
+
+1. Define a new carousel in ``pyvista/doc/source/make_tables.py``, e.g.:
+
+    .. code:: python
+
+        class InstrumentsCarousel(DatasetGalleryCarousel):
+            """Class to generate a carousel of instrument dataset cards."""
+
+            name = 'instrument_carousel'
+            doc = 'Instrument datasets.'
+            badge = CategoryBadge('Instrument', ref='instrument_gallery')
+
+            @classmethod
+            def fetch_dataset_names(cls):
+                return sorted(
+                    (
+                        'guitar',
+                        'trumpet',
+                    )
+                )
+
+   where
+
+   -  ``name`` is used internally to define the name of the generated
+      ``.rst`` file for the carousel
+
+   -  ``doc`` is a short text description of the carousel which will
+      appear in the documentation above
+
+   -  ``badge`` is used to give all datasets in the carousel a reference
+      tag. The ``ref`` argument for the badge should be a new reference
+      target (details below).
+
+   -  ``fetch_dataset_names`` should return a list of any/all dataset names
+      to be included in the carousel. The dataset names should not include
+      any ``load_``, ``download_``, or ``dataset_`` prefix.
+
+2. Add the new carousel class to the the ``CAROUSEL_LIST`` variable defined
+   in ``pyvista/doc/source/make_tables.py``. This will enable the rst to be
+   auto-generated for the carousel.
```

Comment:
```suggestion
1. Add the new carousel class to the the ``CAROUSEL_LIST`` variable defined
   in ``pyvista/doc/source/make_tables.py``. This will enable the rst to be
   auto-generated for the carousel.
```